include/dynd/bytes.hpp

```diff
@@ -89,14 +89,22 @@ class DYND_API bytes {
 
   const bytes operator+(const bytes& rhs)
   {
-      bytes result;
+    bytes result;
 
-      result.resize(size() + rhs.size());
+    result.resize(size() + rhs.size());
 
-      DYND_MEMCPY(result.begin(), begin(), size());
-      DYND_MEMCPY(result.begin() + size(), rhs.begin(), rhs.size());
+    DYND_MEMCPY(result.begin(), begin(), size());
+    DYND_MEMCPY(result.begin() + size(), rhs.begin(), rhs.size());
 
-      return result;
+    return result;
+  }
+
+  bytes &operator+=(const bytes& rhs)
+  {
+    size_t orig_size = size();
+    resize(size() + rhs.size());
+    DYND_MEMCPY(begin() + orig_size, rhs.begin(), rhs.size());
+    return *this;
```

Comment:
It looks like because the string isn't following the design in https://github.com/libdynd/libdynd/blob/master/docs/string-design.md yet, this can't currently be implemented with an exponential growth strategy for efficient amortized concatenation. What thoughts does everyone have on finishing that up?